c10/util/Metaprogramming.h

```diff
@@ -49,6 +49,22 @@ struct infer_function_traits<Result (Args...)> {
 template <typename T>
 using infer_function_traits_t = typename infer_function_traits<T>::type;
 
+/**
+ * make_function_traits: creates a `function_traits` type given a Return type
+ * and a typelist of Argument types
```

Comment:
Implementation looks good. Can you add a usage example to the comment and test cases to Metaprogramming_test.cpp?